flink-runtime/src/main/java/org/apache/flink/streaming/api/datastream/KeyedStream.java

```diff
@@ -358,9 +359,10 @@ public <R> SingleOutputStreamOperator<R> process(
     @Internal
     public <R> SingleOutputStreamOperator<R> process(
             KeyedProcessFunction<KEY, T, R> keyedProcessFunction, TypeInformation<R> outputType) {
-
-        KeyedProcessOperator<KEY, T, R> operator =
-                new KeyedProcessOperator<>(clean(keyedProcessFunction));
+        OneInputStreamOperator operator =
+                isEnableAsyncState()
+                        ? new AsyncKeyedProcessOperator<>(clean(keyedProcessFunction))
```

Comment:
nit: `clean(keyedProcessFunction)` is repeated on both lines - can we user a variable?

Reply:
If we use a variable, a new line will be introduced here, I prefer to keep it as it is.